### ISSUE TITLE
SQL Exception "can't specify target table 'oc_music_albums' for update"

### ISSUE THREAD

**Issue by binarykitchen** [owncloud/music#378]
Happens on owncloud v8.0.3-2 and PHP 5.5.24:

```
{"reqId":"aab4a7d5af6b41c9a42505c0100fb543","remoteAddr":"","app":"no app in context","message":"Error while running background job: An exception occurred while executing 'UPDATE `oc_music_albums` SET `cover_file_id` = NULL WHERE `cover_file_id` IS NOT NULL AND `cover_file_id` IN (SELECT `cover_file_id` FROM `oc_music_albums` LEFT JOIN `oc_filecache` ON `cover_file_id`=`fileid` WHERE `fileid` IS NULL);':\n\nSQLSTATE[HY000]: General error: 1093 You can't specify target table 'oc_music_albums' for update in FROM clause","level":3,"time":"2015-05-09T01:30:15+00:00"}
```

**Comment by binarykitchen** [comment]
Do I really have to add more info? The error is pretty clear, a bad SQL syntax. @RealRancor @MorrisJobke

**Comment by MorrisJobke** [comment]
> Do I really have to add more info? The error is pretty clear, a bad SQL syntax. @RealRancor @MorrisJobke

Thanks. It's an MySQL DB, right?

**Comment by RealRancor** [comment]
> Do I really have to add more info?
It could be useful to use the issue template as the question above where already answered if you had used it :-)

**Comment by binarykitchen** [comment]
I'm tired of using the issue template every time. Wish there is some automatic error reporting tool integrated into owncloud.

Alright, here some more information:
* I'm running owncloud v8.0.3 on PHP 5.5.24 under Ubuntu 14.10
* MySQL 5.5.43 and the php5-mysql module is v5.5.25
* Nginx version 1.8.0

All pretty much the latest version. Hope you can locate and eliminate the bug as it's polluting the log file every day.

**Comment by RealRancor** [comment]
Is your environment changing daily or why don't you fill out the template once and then paste it later in new created issues? :wink:

**Comment by binarykitchen** [comment]
Alright, it took me 20 minutes to fill out the template and hope this is good enough for you guys:

### Steps to reproduce
1. No idea. I just see that line in the log.

### Expected behaviour
That line shouldn't be in the log.

### Actual behaviour
That line appears in the log.

### Server configuration

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.2 LTS
Release:        14.04
Codename:       trusty
```

```
$ uname -a      
Linux binarykitchen.com 3.12.27-xenU #1 SMP Thu Aug 28 23:13:07 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```

Web server:

```
$ nginx -v
nginx version: nginx/1.8.0
```

Database:

```
$ mysql --version
mysql  Ver 14.14 Distrib 5.5.43, for debian-linux-gnu (x86_64) using readline 6.3
```

PHP version:

```
$ php -v
PHP 5.5.25-1+deb.sury.org~trusty+2 (cli) (built: May 24 2015 15:45:37) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2015 Zend Technologies
    with Zend OPcache v7.0.4-dev, Copyright (c) 1999-2015, by Zend Technologies
```

ownCloud version:
ownCloud 8.0.3.4 (stable) 

Updated from an older ownCloud or fresh install:
Fresh install via apt-get

```
$ sudo -u www-data php ./occ app:list
Enabled:
 - activity
 - calendar
 - documents
 - files
 - files_encryption
 - files_locking
 - files_pdfviewer
 - files_sharing
 - files_texteditor
 - files_trashbin
 - files_versions
 - files_videoviewer
 - galleryplus
 - mozilla_sync
 - music
 - provisioning_api
 - tasks
 - templateeditor
 - user_webdavauth
 - videos
Disabled:
 - external
 - files_external
 - firstrunwizard
 - gallery
 - user_external
 - user_ldap
```

The content of config/config.php: (Without sensitive data)

```
$ cat config.php
<?php
$CONFIG = array (
  'instanceid' => 'oc02ca513338',
  'passwordsalt' => 'xxx',
  'datadirectory' => '/var/www/owncloud/data',
  'dbtype' => 'mysql',
  'version' => '8.0.3.4',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'owncloud',
  'dbpassword' => 'xxx',
  'installed' => true,
  'forcessl' => true,
  'loglevel' => 1,
  'logfile' => '/var/log/owncloud.log',
  'theme' => '',
  'maintenance' => false,
  'trusted_domains' => 
  array (
    0 => 'xxx.yyy.com',
  ),
  'mail_smtpmode' => 'smtp',
  'secret' => 'xxx',
  'forceSSLforSubdomains' => true,
  'mail_smtpsecure' => 'tls',
  'mail_smtphost' => '127.0.0.1',
  'mail_smtpport' => '587',
  'mail_smtpauth' => 1,
  'mail_smtpname' => 'xxx',
  'mail_smtppassword' => 'xxx',
  'mail_smtpauthtype' => 'PLAIN',
);
```

Are you using external storage, if yes which one:
None

### Client configuration
Browser:
Firefox 38

Operating system:
Ubuntu 15.04

### Logs

#### Web server error log
nginx error log is empty since a couple of days, hence irrelevant.

#### ownCloud log (data/owncloud.log)
```
{"reqId":"aab4a7d5af6b41c9a42505c0100fb543","remoteAddr":"","app":"no app in context","message":"Error while running background job: An exception occurred while executing 'UPDATE `oc_music_albums` SET `cover_file_id` = NULL WHERE `cover_file_id` IS NOT NULL AND `cover_file_id` IN (SELECT `cover_file_id` FROM `oc_music_albums` LEFT JOIN `oc_filecache` ON `cover_file_id`=`fileid` WHERE `fileid` IS NULL);':\n\nSQLSTATE[HY000]: General error: 1093 You can't specify target table 'oc_music_albums' for update in FROM clause","level":3,"time":"2015-05-09T01:30:15+00:00"}
```

#### Browser log
Irrelevant

**Comment by MorrisJobke** [comment]
Fixed with #372
